### PR TITLE
[Improve][Transform] Support errorHandleWay on jsonpath transform

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Improve][Transform] Ignore error with jsonpath transform

### Does this PR introduce _any_ user-facing change?

Add new config item

### How was this patch tested?

Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by FuYouJ** [comment]
good idea

File: docs/en/transform-v2/jsonpath.md [hunk 1]
```diff
@@ -51,6 +60,14 @@ Support SeatunnelDateType
 
 > Jsonpath
 
+#### column_error_handle_way [Enum]
```
Comment (Hisoka-X) [inline docs/en/transform-v2/jsonpath.md:63]: Add a `column_error_handle_way` config in example? And please share more about priority between `column_error_handle_way` and `row_error_handle_way`.

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/common/AbstractCatalogSupportTransform.java [hunk 2]
```diff
@@ -22,21 +22,47 @@
...
 
     @Override
     public SeaTunnelRow map(SeaTunnelRow row) {
-        return transformRow(row);
+        try {
+            return transformRow(row);
+        } catch (ErrorDataTransformException e) {
+            if (e.getErrorHandleWay() != null) {
+                ErrorHandleWay errorHandleWay = e.getErrorHandleWay();
+                if (errorHandleWay.allowSkip() || errorHandleWay.allowSkipThisRow()) {
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/common/AbstractCatalogSupportTransform.java]:
```suggestion
                if (errorHandleWay.allowSkipThisRow()) {
```
The `errorHandleWay.allowSkip()` always be false.

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java [hunk 3]
```diff
@@ -16,15 +16,217 @@
...
+                new JsonPathTransform(JsonPathTransformConfig.of(config), table);
+
+        CatalogTable outputTable = transform.getProducedCatalogTable();
+        SeaTunnelRow outputRow;
+        try {
+            outputRow = transform.map(new SeaTunnelRow(new Object[] {"{\"f2\": 1}"}));
+            Assertions.fail("should throw exception");
+        } catch (Exception e) {
+            // ignore
+        }
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java]: The `Assertions.fail("should throw exception");` will be catch too?
I think you can use `Assertions.assertThrows()`

**Comment by hailin0** [comment]
image

File: seatunnel-core/seatunnel-spark-starter/seatunnel-spark-starter-common/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java [hunk 4]
```diff
@@ -102,11 +104,23 @@ public List<DatasetTableInfo> execute(List<DatasetTableInfo> upstreamDataStreams
         }
         ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         DatasetTableInfo input = upstreamDataStreams.get(0);
+
+        Map<String, DatasetTableInfo> outputTables =
+                upstreamDataStreams.stream()
```
Comment (hailin0) [inline seatunnel-core/seatunnel-spark-starter/seatunnel-spark-starter-common/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java:109]: fix spark bugs when jsonpath e2e

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java [hunk 5]
```diff
@@ -85,12 +88,22 @@ public List<DataStreamTableInfo> execute(List<DataStreamTableInfo> upstreamDataS
             return upstreamDataStreams;
         }
         DataStreamTableInfo input = upstreamDataStreams.get(0);
+        Map<String, DataStreamTableInfo> outputTables =
```
Comment (hailin0) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java:91]: fix spark bugs when jsonpath e2e

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java (hunk 3 above)
Comment (hailin0) [reply seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java]: Assertions.fail throw Error will not be Exception caught

File: docs/en/transform-v2/jsonpath.md (hunk 1 above)
Comment (hailin0) [reply docs/en/transform-v2/jsonpath.md:63]: fixed

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/common/AbstractCatalogSupportTransform.java (hunk 2 above)
Comment (hailin0) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/common/AbstractCatalogSupportTransform.java]: fixed

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java (hunk 3 above)
Comment (hailin0) [reply seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java]: fixed

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java [hunk 6]
```diff
@@ -16,15 +16,212 @@
...
+import org.apache.seatunnel.transform.jsonpath.JsonPathTransformConfig;
 
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
 public class JsonPathTransformFactoryTest {
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java]:
```suggestion
public class JsonPathTransformTest {
```

Comment (hailin0) [reply seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java]: fixed

**Review by Hisoka-X — APPROVED** [review]

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

**Pull request merged**